indico/modules/events/registration/fields/simple.py

```diff
@@ -388,3 +389,28 @@ def _indico_email(value):
                 raise ValidationError(_('Invalid email address'))
 
         return _indico_email
+
+
+class PictureField(FileField):
+    name = 'picture'
+    setup_schema_fields = {
+        'min_picture_size': fields.Integer(load_default=0, validate=validate.Range(0, 1200)),
+    }
+
+    def get_validators(self, existing_registration):
+        def _picture_size_and_type(value):
+            if not value:
+                return
+            file = File.query.filter(File.uuid == value, ~File.claimed).first() if value else None
+            if not file:
+                raise ValidationError('Invalid file')
+            if file.content_type not in ('image/png', 'image/jpg', 'image/gif', 'image/webp'):
+                raise ValidationError(_('This field can accept only .jpg, .png, .gif and .webp picture formats.'))
+            min_picture_size = self.form_item.data.get('min_picture_size')
+            if min_picture_size:
+                with Image.open(file.open()) as picture:
```

Comment:
with file.open() as f, Image.open(f) as img: ...

not sure if this works, if not use two separate nested with blocks. But the File should be opened using a context manager just like you'd do for a normal file on disk